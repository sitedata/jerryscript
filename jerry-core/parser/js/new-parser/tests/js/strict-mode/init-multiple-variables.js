/* This test contains multiple variable initialization tests. */

next_statement;

var l, i, s;

function f() {}
function g() {}
function h() {}
function i() {}
function j() {}
function k() {}
function l() {}
function m() {}

f+g+h+i+j+k+l+m;

next_statement;

(function (l, i, c) {

var a,b;

function f() {}
function g() {}
function h() {}
function i() {}
function j() {}
function k() {}
function l() {}
function m() {}

f+g+h+i+j+k+l+m;
});

next_statement;
