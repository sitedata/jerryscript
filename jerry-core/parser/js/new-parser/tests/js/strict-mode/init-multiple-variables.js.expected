
--- Script parsing start ---

  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK

--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---

  [  1] CBC_ADD_TWO_LITERALS idx:4->var_ident(f) idx:6->var_ident(g)
  [  1] CBC_ADD_RIGHT_LITERAL idx:8->var_ident(h)
  [  1] CBC_ADD_RIGHT_LITERAL idx:10->var_ident(i)
  [  1] CBC_ADD_RIGHT_LITERAL idx:12->var_ident(j)
  [  1] CBC_ADD_RIGHT_LITERAL idx:14->var_ident(k)
  [  1] CBC_ADD_RIGHT_LITERAL idx:16->var_ident(l)
  [  1] CBC_ADD_RIGHT_LITERAL idx:18->var_ident(m)
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK

--- Function parsing start ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 0
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 0
  Const literal range end: 0
  Literal range end: 0

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---

  [  1] CBC_ADD_TWO_LITERALS idx:5->var_ident(f) idx:7->var_ident(g)
  [  1] CBC_ADD_RIGHT_LITERAL idx:9->var_ident(h)
  [  1] CBC_ADD_RIGHT_LITERAL idx:11->var_ident(i)
  [  1] CBC_ADD_RIGHT_LITERAL idx:13->var_ident(j)
  [  1] CBC_ADD_RIGHT_LITERAL idx:15->var_ident(k)
  [  1] CBC_ADD_RIGHT_LITERAL idx:17->var_ident(l)
  [  1] CBC_ADD_RIGHT_LITERAL idx:19->var_ident(m)
  [  0] CBC_POP

Final byte code dump:

  Maximum stack depth: 14
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 3
  Register range end: 13
  Identifier range end: 13
  Const literal range end: 13
  Literal range end: 21

   0 : CBC_INITIALIZE_VAR idx:5(reg)->var_ident(f) idx:13(lit)->function
   3 : CBC_INITIALIZE_VAR idx:6(reg)->var_ident(g) idx:14(lit)->function
   6 : CBC_INITIALIZE_VAR idx:7(reg)->var_ident(h) idx:15(lit)->function
   9 : CBC_INITIALIZE_VAR idx:8(reg)->var_ident(i) idx:16(lit)->function
  12 : CBC_INITIALIZE_VAR idx:9(reg)->var_ident(j) idx:17(lit)->function
  15 : CBC_INITIALIZE_VAR idx:10(reg)->var_ident(k) idx:18(lit)->function
  18 : CBC_INITIALIZE_VAR idx:11(reg)->var_ident(l) idx:19(lit)->function
  21 : CBC_INITIALIZE_VAR idx:12(reg)->var_ident(m) idx:20(lit)->function
  24 : CBC_ADD_TWO_LITERALS idx:5(reg)->var_ident(f) idx:6(reg)->var_ident(g)
  27 : CBC_ADD_RIGHT_LITERAL idx:7(reg)->var_ident(h)
  29 : CBC_ADD_RIGHT_LITERAL idx:8(reg)->var_ident(i)
  31 : CBC_ADD_RIGHT_LITERAL idx:9(reg)->var_ident(j)
  33 : CBC_ADD_RIGHT_LITERAL idx:10(reg)->var_ident(k)
  35 : CBC_ADD_RIGHT_LITERAL idx:11(reg)->var_ident(l)
  37 : CBC_ADD_RIGHT_LITERAL idx:12(reg)->var_ident(m)
  39 : CBC_POP
  40 : CBC_RETURN_WITH_BLOCK

Byte code size: 41 bytes

--- Function parsing end ---

  [  1] CBC_PUSH_LITERAL idx:20->function
  [  0] CBC_POP_BLOCK
  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK

Final byte code dump:

  Maximum stack depth: 1
  Flags: [small_lit_enc]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 10
  Const literal range end: 10
  Literal range end: 19

   0 : CBC_DEFINE_VARS from: 0 to: 0
         idx:0(ident)->var_ident(s)
   2 : CBC_INITIALIZE_VARS from: 1 to: 8
         idx:1(ident)->var_ident(f) = idx:10(lit)->function
         idx:2(ident)->var_ident(g) = idx:11(lit)->function
         idx:3(ident)->var_ident(h) = idx:12(lit)->function
         idx:4(ident)->var_ident(i) = idx:13(lit)->function
         idx:5(ident)->var_ident(j) = idx:14(lit)->function
         idx:6(ident)->var_ident(k) = idx:15(lit)->function
         idx:7(ident)->var_ident(l) = idx:16(lit)->function
         idx:8(ident)->var_ident(m) = idx:17(lit)->function
  13 : CBC_PUSH_LITERAL idx:9(ident)->ident(next_statement)
  15 : CBC_POP_BLOCK
  16 : CBC_ADD_TWO_LITERALS idx:1(ident)->var_ident(f) idx:2(ident)->var_ident(g)
  19 : CBC_ADD_RIGHT_LITERAL idx:3(ident)->var_ident(h)
  21 : CBC_ADD_RIGHT_LITERAL idx:4(ident)->var_ident(i)
  23 : CBC_ADD_RIGHT_LITERAL idx:5(ident)->var_ident(j)
  25 : CBC_ADD_RIGHT_LITERAL idx:6(ident)->var_ident(k)
  27 : CBC_ADD_RIGHT_LITERAL idx:7(ident)->var_ident(l)
  29 : CBC_ADD_RIGHT_LITERAL idx:8(ident)->var_ident(m)
  31 : CBC_POP_BLOCK
  32 : CBC_PUSH_LITERAL idx:9(ident)->ident(next_statement)
  34 : CBC_POP_BLOCK
  35 : CBC_PUSH_LITERAL idx:18(lit)->function
  37 : CBC_POP_BLOCK
  38 : CBC_PUSH_LITERAL idx:9(ident)->ident(next_statement)
  40 : CBC_POP_BLOCK
  41 : CBC_RETURN_WITH_BLOCK

Byte code size: 42 bytes

Script parsing successfully completed. Total byte code size: 99 bytes

--- Script parsing end ---

